src/engine/Match_rules.mli

```diff
@@ -14,6 +14,7 @@ val check :
   match_hook:(string -> Pattern_match.t -> unit) ->
   timeout:float ->
   timeout_threshold:int ->
+  ?dependency_match_table:Match_dependency.dependency_match_table ->
```

Comment:
Can you add a comment for this new field, and why we need to pass it around, why we can't post-process the match instead?